File: DFS, BFS/1697.py
#숨바꼭질
from collections import deque
def bfs(x):
    graph[x] = 1
    queue = deque()
    queue.append(x)
    while queue:
        x = queue.popleft()
        for nx in (x-1, x+1, 2*x):
            if 0<=nx<100001 and graph[nx] == 0:
                graph[nx] = graph[x] + 1
                queue.append(nx)
    return graph[k]-1
graph = [0 for _ in range(100001)]
n, k = map(int, input().split())
print(bfs(n))

File: DFS, BFS/1697.js
// 숨바꼭질
let input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split(" ");

let [n, k] = input.map((v) => Number(v));
let graph = Array(100001).fill(0);

bfs();
console.log(graph[k]);

function bfs() {
  if (n === k) return;
  let q = [];
  q.push(n);
  while (q.length) {
    let x = q.shift();
    for (let nx of [x - 1, x + 1, 2 * x]) {
      if (0 <= nx && nx <= 100000 && graph[nx] === 0) {
        graph[nx] = graph[x] + 1;
        q.push(nx);
      }
    }
  }
}
